hugegraph-llm/src/hugegraph_llm/operators/index_op/build_vector_index.py

```diff
@@ -40,8 +40,7 @@ def run(self, context: Dict[str, Any]) -> Dict[str, Any]:
         chunks_embedding = []
         log.debug("Building vector index for %s chunks...", len(context["chunks"]))
         # TODO: use async_get_texts_embedding instead of single sync method
-        for chunk in tqdm(chunks):
-            chunks_embedding.append(self.embedding.get_text_embedding(chunk))
+        chunks_embedding = asyncio.run(get_embeddings_parallel(self.embedding, chunks))
```

Comment:
Avoid calling asyncio.run() inside library code; it breaks when a loop is already running (e.g., FastAPI/Notebook).

Consider: (1) make run() async and let callers decide, or (2) add a sync wrapper that detects an existing loop and uses run_coroutine_threadsafe; only use asyncio.run if no loop exists.